### DISCUSSION TITLE
Starting uwsgi application under Memray

### DISCUSSION THREAD

**Discussion by ghost** [bloomberg/memray#60] [Q&A] [answered]
Our Python application is deployed under uwsgi. However, uwsgi itself uses a native executable as its entry point, which memray cannot run directly.

Is there any way to use memray in conjunction with uwsgi? I have full control over the application code and the uwsgi configuration.

**Comment by godlygeek** [comment, marked as answer]
I'm not terribly familiar with uWSGI, but it should be possible to use [the Memray API](https://bloomberg.github.io/memray/api.html) even with uWSGI. I think if you were to use [uWSGI hooks](https://uwsgi-docs.readthedocs.io/en/latest/Hooks.html#the-hookable-uwsgi-phases), it ought to be possible to create a [Memray `Tracker`](https://bloomberg.github.io/memray/api.html#api-usage) and call its `__enter__` with an `accepting-once` hook, which should be able to get you an output file for that worker process.

**Reply by ghost** [reply]
I don't see a way to call into Python code from uWSGI's hooks, but I'm not terribly familiar with that part of uWSGI either. However, the Memray API is exactly what I'm looking for, and I can create a `Tracker` and call its `__enter__` right before (or after, as the case may be) initializing the application. Thanks!

**Comment by ptitpoulpe** [comment]
@zr40 did you found a way to use `Tracker` to track all threads ? seems I only trace the main one

**Reply by ghost** [reply]
No, sorry. The application I needed to profile is single-threaded.
